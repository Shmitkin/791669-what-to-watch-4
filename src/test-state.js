export const movie = {
  title: `Pulp Fiction`,
  videoPrev: `img/seven-years-in-tibet.mp4`,
  preview: `img/pulp-fiction.jpg`,
  poster: `img/pulp-fiction.jpg`,
  background: `img/pulp-fiction.jpg`,
  genre: `Drama`,
  description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent varius quis lorem id iaculis. Sed eget luctus orci. Donec cursus lectus auctor quam tempus ultricies. Nullam at aliquam massa. Donec mollis quis urna eget condimentum. Mauris at dictum ante. Phasellus maximus massa augue, ut luctus dolor rutrum et. Aliquam faucibus turpis lectus, sed hendrerit dolor malesuada at.`,
  rating: 2.3,
  votes: 293,
  director: `Best Director`,
  starring: `Nicola Cage, Cara Delvin, Joseph Moser`,
  duration: 87,
  release: 2020,
};

export const movies = [
  {
    title: `Seven Years IN Tibet`,
    videoPrev: `img/seven-years-in-tibet.mp4`,
    preview: `img/seven-years-in-tibet.jpg`,
    poster: `img/seven-years-in-tibet.jpg`,
    background: `img/seven-years-in-tibet.jpg`,
    genre: `Family`,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent varius quis lorem id iaculis. Sed eget luctus orci. Donec cursus lectus auctor quam tempus ultricies. Nullam at aliquam massa. Donec mollis quis urna eget condimentum. Mauris at dictum ante. Phasellus maximus massa augue, ut luctus dolor rutrum et. Aliquam faucibus turpis lectus, sed hendrerit dolor malesuada at.`,
    rating: 5.0,
    votes: 1255,
    director: `Best Director`,
    starring: `Nicola Cage, Cara Delvin, Joseph Moser`,
    duration: 24,
    release: 2005,
  },
  {
    title: `Moonrise Kingdom`,
    videoPrev: `img/seven-years-in-tibet.mp4`,
    preview: `img/moonrise-kingdom.jpg`,
    poster: `img/moonrise-kingdom.jpg`,
    background: `img/moonrise-kingdom.jpg`,
    genre: `Horror`,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent varius quis lorem id iaculis. Sed eget luctus orci. Donec cursus lectus auctor quam tempus ultricies. Nullam at aliquam massa. Donec mollis quis urna eget condimentum. Mauris at dictum ante. Phasellus maximus massa augue, ut luctus dolor rutrum et. Aliquam faucibus turpis lectus, sed hendrerit dolor malesuada at.`,
    rating: 4.8,
    votes: 213,
    director: `Best Director`,
    starring: `Nicola Cage, Cara Delvin, Joseph Moser`,
    duration: 79,
    release: 2012,
  },
  {
    title: `Snatch`,
    videoPrev: `img/seven-years-in-tibet.mp4`,
    preview: `img/snatch.jpg`,
    poster: `img/snatch.jpg`,
    background: `img/snatch.jpg`,
    genre: `Comedy`,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent varius quis lorem id iaculis. Sed eget luctus orci. Donec cursus lectus auctor quam tempus ultricies. Nullam at aliquam massa. Donec mollis quis urna eget condimentum. Mauris at dictum ante. Phasellus maximus massa augue, ut luctus dolor rutrum et. Aliquam faucibus turpis lectus, sed hendrerit dolor malesuada at.`,
    rating: 3.0,
    votes: 22,
    director: `Best Director`,
    starring: `Nicola Cage, Cara Delvin, Joseph Moser`,
    duration: 210,
    release: 1998,
  },
];

export const genres = [
  `First`,
  `Second`,
  `Third`,
  `Fourth`,
  `Fifth`
];
