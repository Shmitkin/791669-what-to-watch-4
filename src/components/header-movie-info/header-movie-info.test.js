import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";

import HeaderMovieInfo from "./header-movie-info.jsx";
import {mockStore, movie} from "../../test-state.js";

it(`Should HeaderMovieInfo render correctly`, () => {
  const testStore = configureStore();
  const store = testStore(mockStore);

  const tree = renderer
    .create(
        <Provider store={store}>
          <HeaderMovieInfo
            movie = {movie}
            isMovieDetails = {false}
          />
        </Provider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
