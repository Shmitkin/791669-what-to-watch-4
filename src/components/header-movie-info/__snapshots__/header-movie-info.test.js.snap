// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should HeaderMovieInfo render correctly 1`] = `
<div
  className="movie-card__info"
>
  <div
    className="movie-card__poster "
  >
    <img
      alt="Pulp Fiction poster"
      height="327"
      src="img/pulp-fiction.jpg"
      width="218"
    />
  </div>
  <div
    className="movie-card__desc"
  >
    <h2
      className="movie-card__title"
    >
      Pulp Fiction
    </h2>
    <p
      className="movie-card__meta"
    >
      <span
        className="movie-card__genre"
      >
        Drama
      </span>
      <span
        className="movie-card__year"
      >
        2020
      </span>
    </p>
    <div
      className="movie-card__buttons"
    >
      <button
        className="btn btn--play movie-card__button"
        type="button"
      >
        <svg
          height="19"
          viewBox="0 0 19 19"
          width="19"
        >
          <use
            xlinkHref="#play-s"
          />
        </svg>
        <span>
          Play
        </span>
      </button>
      <button
        className="btn btn--list movie-card__button"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="20"
          viewBox="0 0 19 20"
          width="19"
        >
          <use
            xlinkHref="#add"
          />
        </svg>
        <span>
          My list
        </span>
      </button>
    </div>
  </div>
</div>
`;
